src/oas2/transformers/params.ts

```diff
@@ -179,7 +179,7 @@ export function translateToPathParameter(parameter: PathParameter): IHttpPathPar
 }
 
 function buildSchemaForParameter(
-  param: QueryParameter | PathParameter | HeaderParameter | FormDataParameter | Header,
+  param: (QueryParameter | PathParameter | HeaderParameter | FormDataParameter | Header) & { readOnly?: boolean },
```

Comment:
@XVincentX this is due to a typing problem. I haven't analyzed in thoroughly yet but I guess somewhere `required` is defined as array and elsewhere as boolean. When I add `readOnly`, TS compiler picks different type and complains. 

I'll take a closer look tomorrow.

Reply:
"required is an array but elsewhere its a boolean" 

This is not the same keyword. Inside a Schema Object, required is an array. Parameter Object (not a schema object) required is a boolean. We cannot mix up Schema Objects and Parameter Objects, or get confused about which keyword goes in where. The spec should show us which goes where, even if it is a bloomin tricky read.

Reply:
A more specific note: `readOnly` is only mentioned in one place in the spec, and its in the Schema Object. This probably should not be here.

Reply:
Ok this looks weird. I'm digging into this as well and see if I can help.

Reply:
@karol-maciaszek I do not think this change is required at all, effectively this property is only in the `Schema` object. Tests are indeed passing even without this change.

Reply:
Yes, the code actually copies schema stuff from a param to schema. So `readOnly` is not applicable here.